### PR TITLE
Feature/verify-post

### PR DESCRIPTION
**Pull request by artyom-yurin**
### Description of the Change
Added functions to generate and verify a PoSt

### Benefits
Unblocks StoragePowerActor

### Possible Drawbacks 
All errors are UKNOWN, because there is no complete list of errors
Therefore I used a logger. Maybe it is can be a drawback

### CODE REVIEW

File: test/core/proofs/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+#SPDX - License - Identifier : Apache - 2.0
+#
+
+addtest(proofs_test proofs_test.cpp)
+
+target_link_libraries( proofs_test
+        proofs
+        filecoin_ffi
+        outcome
+        blob
```
Comment (turuslan) [inline test/core/proofs/CMakeLists.txt]: Remove, already linked by proofs

File: core/proofs/proofs.cpp [hunk 2]
```diff
@@ -0,0 +1,471 @@
...
+  using common::Blob;
+  using crypto::randomness::Randomness;
+
+  // ******************
+  // TO CPP CASTED FUNCTIONS
+  // ******************
+
+  Blob<kCommitmentBytesLen> cppCommitment(const uint8_t *src) {
+    Blob<kCommitmentBytesLen> result;
+    for (size_t i = 0; i < kCommitmentBytesLen; i++) {
```
Comment (turuslan) [inline core/proofs/proofs.cpp]: Use `std::copy`

File: test/core/proofs/CMakeLists.txt (hunk 1 above)
Comment (artyom-yurin) [reply test/core/proofs/CMakeLists.txt]: Removed

File: core/proofs/proofs.cpp (hunk 2 above)
Comment (artyom-yurin) [reply core/proofs/proofs.cpp]: Fixed

File: core/proofs/proof_param_provider.cpp [hunk 3]
```diff
@@ -0,0 +1,281 @@
...
+          out.c_str(), boost::beast::file_mode::append, ec_file);
+      // Receive the HTTP response
+      http::read(stream, buffer, res);
+      // Gracefully close the socket
+      beast::error_code ec;
+      stream.socket().shutdown(tcp::socket::shutdown_both, ec);
+
+      // not_connected happens sometimes
+      // so don't bother reporting it.
+      //
```
Comment (Alexey-N-Chernyshov) [inline core/proofs/proof_param_provider.cpp]: extra comment string

Comment (artyom-yurin) [reply core/proofs/proof_param_provider.cpp]: Fixed

**Pull request merged**